### ISSUE TITLE
"dim" argument for .unfold method doesn't work

### ISSUE THREAD

**Issue by seungwonpark** [pytorch/pytorch#18793]
## 🐛 Bug

When using `.unfold` method of `torch.Tensor`, keyword argument `dim` doesn't work: only `dimension` works.

## To Reproduce

```
>>> x = torch.randn(34)
>>> y = x.unfold(dim=0, size=10, step=5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unfold() missing 3 required positional argument: "dimension", "size", "step"
>>> y = x.unfold(dimension=0, size=10, step=5)
>>> y.shape
torch.Size([5, 10])
```

## Environment

```
Collecting environment information...
PyTorch version: 1.0.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: CentOS Linux 7 (Core)
GCC version: (GCC) 5.3.0
CMake version: version 3.13.3

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: GeForce GTX 1080
GPU 1: GeForce GTX 1080

Nvidia driver version: 410.79
cuDNN version: Probably one of the following:
/usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudnn.so.7
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.5
/usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.7

Versions of relevant libraries:
[pip3] numpy==1.15.4
[pip3] torch==1.0.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.2.1
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.1                      144  
[conda] mkl-service               1.1.2            py37he904b0f_5  
[conda] mkl_fft                   1.0.6            py37hd81dba3_0  
[conda] mkl_random                1.0.2            py37hd81dba3_0
```

**Comment by vishwakftw** [comment]
Related to https://github.com/pytorch/pytorch/issues/8698

**Comment by ezyang** [comment]
Doc fix is in #19020; maybe we want to make the function accept both keywords, but for now fixing the docs is the easiest thing to do.
